Paper: Accelerating the Advancement of Data Science Education (#478)
scipy-conference/scipy_proceedings: papers/Amal_Bhatnagar/NationalAdvancementofDataScience.rst

:author: Anthony Suen
:email: anthonysuen@berkeley.edu 
:institution: University of California, Berkeley

:author: Alan Liang
:email: alanliang@berkeley.edu
:institution: University of California, Berkeley

:author: Amal Bhatnagar
:email: amalbhatnagar@berkeley.edu
:institution: University of California, Berkeley

------------------------------------------------
National Advancement of Data Science Education
------------------------------------------------

Data science is a burgeoning field that is quickly being adopted across
all domains and sectors. Data Science education initiatives have been

> growing rapidly, but also largely simultaneous and in an uncoordinated

Reviewer: ```suggestion
growing rapidly, but also largely simultaneously and in an uncoordinated
```

Author: Edited, thank you!

manner. Programs are often developed and implemented in silos, often
leading to duplications of efforts and differences in pedagogical
approaches and course quality. Furthermore, while a number of curriculum
guidelines for degrees in data science have been proposed, opportunities
for engaging in pedagogical exchanges and sharing resources remain rare.
Without a common knowledge base of resources and platform for
undergraduate support, many institutions have encountered difficulties
in setting up data science curricula, specifically pedagogical and
infrastructural barriers. Additionally, data science education in many
universities has been focused on graduate programs, with very few
undergraduate ones in place.

Stakeholders around the country previously identified the necessity of
gathering data science enthusiasts and discussing its implementation in
institutions. The importance of defining data science curriculum
guidelines has been the subject of numerous workshop and meetings such

> as the “Workshop on Theoretical Foundations of Data Science,” “The Park

Reviewer: Please, include links, or references if available, to these workshops -- where can a reader find more information?

City Math Institute 2016 Summer Undergraduate Faculty Program,” and
“Envisioning the Data Science Discipline: The Undergraduate
Perspective.” While these workshops produced comprehensive guidelines on
the structure of the programs, the actual content and teaching
modalities remain unclear.

On July 16th-July 19th 2018, the Division of Data Sciences at the UC
Berkeley hosted a Workshop on Undergraduate Data Science Pedagogy and
Practice. As a pioneer in undergraduate data science education, UC
Berkeley has developed a comprehensive set of open-license and
open-source resources that range from teaching materials and curriculum

> to its system infrastructure and share them at the workshop. With

Reviewer: ```suggestion
to its system infrastructure that were shared at the workshop. With
```

Author: Edited

support from the NSF, the Berkeley organizers put out a call for

> interested professors and instructors across the country 41 instructors

Reviewer: ```suggestion
interested professors and instructors across the country. The workshop brought together 41 instructors
```
The organizers put out a call, and the 41 participants came to the workshop -- do I understand that correctly?

Author: Yes, that is correct. I edited it to be more clear. Thanks.

from a variety of higher-education institutions that hugely vary in
terms of size, location, and experience with data science education.

The workshop sought to 1) Examine the *Foundations of Data Science (Data
8)* course, and discuss the course content and pedagogical methods
employed behind this innovative and accessible curriculum, 2) Build a
national community of practice around undergraduate Data Science
education, 3) Showcase the infrastructural platform that Berkeley has
developed for its courses that involve data science, and empower
participants to use its many open-source components and 4) Engage
workshop participants to identify challenges and needs for moving Data

> Science education forward at their institutions participants

Reviewer: ```suggestion
Science education forward at their institutions. Participants
```

Author: Edited.

collaborated to develop a common vision for expanding and moving forward
the cause of Data Science Pedagogy nationally. These four items were

> broken down core topics including *1)* **Integrating and Serving across

Reviewer: It would help readability to break these out into bullet points.

Author: Edited

> broken down core topics including *1)* **Integrating and Serving across

Reviewer: ```suggestion
broken down into core topics including *1)* **Integrating and Serving across
```

Author: Edited

institutions face from implementing such an infrastructure, and the
possible solution of generating a regional or nationwide educational
cyber-infrastructure for sustainable and scalable benefits*. *3)*
**Integrating Modular DS content into teaching including Ethical
Content** *properly integrate discussions and critical thinking about
ethics into data science courses, and how such an integration could
result in new courses that will better complement the overall scope of
the educational field.*

1) **Integrating and Serving across Disciplines using Computation**
-------------------------------------------------------------------

*Create a space for Data Science to exist as a cross-campus endeavor and
engage faculty in different departments.*

Within higher education, data science can touch all different genres and
disciplines of academia. The proliferation of affordable computational
capacity, migration of publishing channels to the internet, advanced
sensing technology, and other data collection methods has led to the
possibility of data science in almost every area of scientific endeavor.
Current examples of data science being integrated into other disciplines

> include randomized controlled trials in Development Economics published

Reviewer: References to these examples would be good if they are available.

through open data repositories and the integration between law and data
technologies.

The first step toward creating a successful Data Science program within
an institution comes from the successful implementation of an
introductory-level data science course, as utilization existing
introductory computer science and statistics courses in place of a

> foundational data science course slows students learning and limits the

Reviewer: ```suggestion
foundational data science course slows students' learning and limits the
```

Author: Edited

questions.

As for its implementation, Computer Science and Statistics fields may be
a traditional base but the core foundational facilitators must find
connections to other departments and stakeholders. In fielding an
entry-level data science course and fitting it into the curriculum, it’s
crucial to engage with faculty across a variety of disciplines in an
inclusive, supportive way and a spirit of partnership. A full
implementation will involve enriching the teaching of the entry-level
course, injecting computational content into existing courses, and
helping to reframe methods courses within the disciplines so they can
build off of integrative interdisciplinary general training in data
science.

The Data Science students having classes from a field of application
will create a space to apply methods learned in their core classes, and
opportunities to learn theoretical methods that Data Science may be
lacking. Having other departments engage with a Data Science major or
minor will give those departments a way to grow their offerings and
create possibilities for curricular innovation. The Berkeley model
provided a number of illustrations of cross-campus collaboration

> including introductory Data Science course is accompanied by a series of

Reviewer: ```suggestion
including introductory Data Science courses accompanied by a series of
```

Author: Edited

smaller, applied “Connector” courses which give students a flavor of how
data science may be carried out within a given domain. The Data Science
student teams have also supported the creation of data science content

> for inserting in other type of (usually non data science) courses in

Reviewer: ```suggestion
for inserting in other types of (usually non data science) courses in
```

Author: Edited

self-contained “Modules” that can illustration aspects of data science
to a different audience. The Berkeley Data Science program has also
undertaken a series of summer “bootcamp” style workshops open to all

> faculty to encourage faculty to engage and innovate their curriculum.

Reviewer: ```suggestion
faculty to encourage them to engage and innovate their curricula.
```
Please clarify: were these bootcamps open to faculty, but also to the grad students and postdocs mentioned in the next sentence? Perhaps should say "aimed at faculty" rather than just "open to faculty".

Author: They were aimed at faculty (edited to make it more clear). Grad students and postdocs are mentioned to add another way how data science has been implemented

Finally, at a research university such as Berkeley, there has been
considerable success due to graduate students or postdocs, who may be
leading the way for faculty, in adapting new data science methods to
different disciplines.

**2) Common Educational Cyber-Infrastructure**
----------------------------------------------

> Implementation of a data science course in a scalable way requires

Reviewer: While reading this section, I kept thinking back to [this paper from SciPy 2018](http://conference.scipy.org/proceedings/scipy2018/anthony_suen_laura_noren_alan_liang_andrea_tu.html). No wonder, the first author is also Anthony Suen! Please cite it in this section.

computation is not ideal, as it is harder to manage when the number of
students increases. Additionally, as the number of courses that require
such infrastructure increases, local computation would become too time
consuming. For many small institutions and universities, this proves a
difficult task that can be a barrier to innovation in curriculum and
course delivery. As a result, development of regional or national cloud
based computing solutions that can serve array individual educational
institutions is needed.

*Fund and pilot a regional or national data science hub for education
that will expand access and encourage innovation in data science
education.*

Educational cyber-infrastructure is different than research
cyber-infrastructure due to differences in its goals, resource needs,
deployment timelines, cost and pricing of models, and broad access
mandate. Educational infrastructure is deployed for a relatively low
resource use by a large number of relatively unsophisticated users.
Making the infrastructure accessible means making it easy to use both by
instructors and students, and potentially integrating it into existing

> campus Learning Management Systems (LMS), eg Canvas. For institutions

Reviewer: I am not familiar with Canvas. Would you please include a link or a reference?

world and the system holds all the necessary material. It also makes
teaching data management and analysis and allowing the ability to have
instructors illustrate the visualization of data easier.

However, the adoption costs of cyber-infrastructure is high and
problematic, especially for smaller institutions. While the component
cost for hardware and software are going down through virtualization
(cloud), human talent is hard to acquire. For many institutions, the
ability to setup the necessary support systems for JupyterHub or other
infrastructure is beyond the expertise of a single course instructor.
Even qualified instructors may not have the capacity to take on such a
task, as their time is required for equally important tasks of planning
lesson outlines and curriculum. Institutional IT staff members would
also be required to go through additional training if they were assigned
the task, and the trainings required would vary across institutions to
better fit the differing needs and implementations of the data science
courses. Thus overall startup costs are expensive, and the long term
sustainability for maintaining a educational cyber-infrastructure would

> come with too many question marks for many institutions faculty to make

Reviewer: ```suggestion
come with too many question marks for many institutions' faculty to make
```

Author: Edited

implementation a priority.

Autograding is essential to the scalability of data science education
and alleviates substantial work for large classes at UC Berkeley, such
as *Data 8:* *Foundations of Data Science* and *Data 8X*, its massive

> open online course, or MOOC, version, which see more than 1,500 students

Reviewer: ```suggestion
open online course, or MOOC, version, which sees more than 1,500 students
```

across certain locations might lead to partnerships across and within
institutions, allowing for successful techniques to be communicated
across all partners and similar curriculum modeling to exist for
consistency.

The successful formation of a national educational cyber-infrastructure
will allow for data science courses to be supported at institutions and
universities under a cost efficient structure. Stakeholders would no
longer need to implement their own system, but instead could go through
training and onboarding for a national system that will be easy to use
and consistent between institutions. If such a process could be
undertaken, the ability to host data science courses for undergraduate
university-level student will be readily accessible to schools.

**3) Guidelines for Creating and Incorporating Modular DS Content**
--------------------------------------------------------------------

There are two main concerns when modularizing data science content:
*Having just one introductory data science class is not enough to
warrant an entire data science curricula, and creating a sustainable
model that supports the data science curricula.*

> Implementation and integrating the new course to fit in the overall

Reviewer: ```suggestion
Implementating and integrating the new course to fit in the overall
```

Author: Edited

Reviewer 2: "Implementing"

academic curriculum is key for a seamless student experience. Because
data science serves functions in a vast array of interdisciplinary
fields of study, the ability to modify the introductory course and

> tailor it to fit in with the current institution curriculum will go a

Reviewer: ```suggestion
tailor it to fit in with the current institutional curriculum will go a
```

Author: Edited

platform with resources. They would adopt these resources and
personalize them on their own curriculums to help their students. A
national data science hub for education would bring together these
institutions and innovate the data science education. Universities would
encourage students to use such data-driven methodologies not just in an
institutional setting but also in their professional careers afterwards.
We believe our methodology will guide our commitment to work together,
structure our cross-campus collaboration, and target grant writing to
support these initiatives.

**Works Cited**
---------------

Baumer, Benjamin S., Daniel T. Kaplan, and Nicholas J. Horton. 2017.
*Modern Data Science with R*. Chapman & Hall.
`*http://mdsr-book.github.io* <http://mdsr-book.github.io>`__

Bowker, Geoffrey C. and Susan Leigh Star. 2000. *Sorting Things Out:
Classification and Its Consequences*. Cambridge: MIT Press.

Desrosières, Alain. 1998. *The Politics of Large Numbers: A History of
Statistical Reasoning*. Cambridge: Harvard University Press.

Eubanks, Virginia. 2018. *Automating Inequality: How High-Tech Tools
Profile, Police, and Punish the Poor*. New York: St. Martin’s.

> Hacking, Ian. 1996. Normal People. Pp. 59-71 in David Olson and Nancy

Reviewer: Not cited in the text.

Torrance, eds., *Modes of Thought: Explorations in Culture and
Cognitions*. Cambridge: Cambridge University Press.

> Hicks, Marie. 2017. *Programmed Inequality: How Britain Discarded Women

Reviewer: Not cited in the text.

Technologists and Lost its Edge in Computing*. Cambridge: MIT Press.

Latour, Bruno. 1990. Technology is society made durable. *The
Sociological Review* 38(1, supplement): 103-131.

> Light, Jennifer S. 1999. When computers were women. *Technology and

Reviewer: Not cited in the text.

Author: Thanks for the input on the citations. We are currently working on editing the paper on another platform and will upload the revised version to GitHub soon.

Culture* 40(3): 455-483.
`*https://www.jstor.org/stable/25147356* <https://www.jstor.org/stable/25147356>`__

> MacKenzie, Donald A. 1981. *Statistics in Britain: 1865-1930; The Social

Reviewer: not cited in the text
